pkg/controller/prometheus.go

```diff
@@ -161,20 +163,34 @@ func (c *Controller) checkAndCreatePrometheusAddlConfig(ctx context.Context, ten
 			return err
 		}
 	} else {
-		var scrapeConfigs []configmaps.ScrapeConfig
+		var scrapeConfigs, exceptedScrapeConfigs []configmaps.ScrapeConfig
 		err := yaml.Unmarshal(secret.Data[miniov2.PrometheusAddlScrapeConfigKey], &scrapeConfigs)
 		if err != nil {
 			return err
 		}
-		// Check if the scrape config is already present
-		hasScrapeConfig := false
+		// get other scrape configs
 		for _, sc := range scrapeConfigs {
-			if sc.JobName == tenant.PrometheusOperatorAddlConfigJobName() {
-				hasScrapeConfig = true
-				break
+			if !strings.HasPrefix(sc.JobName, tenant.PrometheusOperatorAddlConfigJobName()) {
+				exceptedScrapeConfigs = append(exceptedScrapeConfigs, sc)
 			}
 		}
-		if !hasScrapeConfig {
+		exceptedScrapeConfigs = append(exceptedScrapeConfigs, promCfg.ScrapeConfigs...)
+		updateScrapeConfig := false
+		if len(scrapeConfigs) != len(exceptedScrapeConfigs) {
+			updateScrapeConfig = true
+		} else {
+			for i := range scrapeConfigs {
+				if scrapeConfigs[i].JobName != exceptedScrapeConfigs[i].JobName ||
+					scrapeConfigs[i].MetricsPath != exceptedScrapeConfigs[i].MetricsPath ||
```

Comment:
Please add some comment that the generation of the scrape configuration always creates a new bearer token, so it shouldn't be included in this check, because it will always result in it being not equal.

How do we make sure the bearer token is updated when the tenant gets assigned other credentials (different access/secret key). This will result in another bearer token (but nothing else), so it will decide that the scape configuration hasn't changed, but Prometheus won't be able to access MinIO with the old token.